doc/contribute/guidelines.rst

```diff
@@ -554,6 +570,38 @@ exceptions:
 * Avoid using binary literals (constants starting with ``0b``).
 * Avoid using non-ASCII symbols in code, unless it significantly improves
   clarity, avoid emojis in any case.
+* Use proper capitalization of nouns in code comments (e.g. ``UART`` and not
+  ``uart``, ``CMake`` and not ``cmake``).
+
+Beyond C code, the following coding style rules apply to other types of files:
+
+* CMake
+
+  * Indent with spaces, indentation is two spaces.
+  * No space between commands (e.g. ``if``) and the corresponding opening
+    bracket (e.g. ``(``).
+
+* Devicetree
+
+  * Indent with tabs.
+  * Follow the Devicetree specification conventions and rules.
+  * Use dashes (``-``) as word separators for node and property names.
+  * Use underscores (``_``) as word separators in node labels.
+  * Leave a single space on each side of the equal sign (``=``) in property
+    definitions.
+  * No empty lines before a dedenting ``};``.
+  * Single empty line separating nodes at the same hierarchy level.
+
+* Kconfig
+
+  * Line length of 100 columns or fewer.
```

Comment:
```suggestion
```

Reply:
Why? This makes sense in the context of Kconfig, because in Devicetree it is not enforced unless I am mistaken?

Reply:
Because I was under the impression that the enclosing section, "Coding style", already applies to everything? Why would this need to be repeated for Kconfig? And why is it not repeated for CMake, then?

Reply:
Reading this again it looks like "Coding Style" is probably meant to read "C coding style", and that you probably didn't mean to fold Kconfig/devicetree/CMake "under" this section as this has really nothing to do with C, and can mistakenly lead people like myself to then interpret "Coding guidelines" as being "General style guidelines" (and e.g. interpret the bullet "In the documentation, longer lines for URL references are an allowed exception." as applying to RST files, whereas I guess the intent was to apply to *Doxygen* documentation?)

Related(ish): The "Coding guidelines" section is pretty buried in the document.

Reply:
What you say is correct and would benefit from a cleanup, but I am trying really really hard to limit the scope of this PR. If I start now rewriting the Coding Style section I will open a can of worms that I would rather leave closed in this iteration, we can then do another pass cleaning up.